main.ml

```diff
@@ -155,11 +155,15 @@ let main force dir =
   let project = Dune_project.parse () in
   let old_opam_files = get_opam_files () in
   let packages = Dune_project.packages project in
-  Paths.iter (fun path _ ->
-      (* prevent `dune describe opam-files` crashing when there is a opam file `*.opam`
-         * and its package description is missing in dune-project file. *)
-      if not (Paths.mem path packages) then Sys.remove path)
-    old_opam_files;
+
+  (* some dune project file has no package description
+   * and avoid removing all the opam files *)
+  if not (Paths.is_empty (Dune_project.packages project)) then (
```

Comment:
`Dune_project.packages project` is computed and bound to `packages` just a few lines above :-)
```suggestion
  if not (Paths.is_empty packages) then (
```